group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -983,27 +1085,81 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> consumerGr
         return new CoordinatorResult<>(records, response);
```

Comment:
Don't we need to also force the step 3.? If we don't do it, we don't write the current assignment record for the new member and we don't reconcile him.

Reply:
Yes, I am now building the replacing static member with the same set of assignments (target/pending). This forces it directly to have the current assignment record for the new member.